contrib/basicauth/nginx/registry.conf

```diff
@@ -0,0 +1,17 @@
+# Docker registry proxy for api version 2
+
+upstream docker-registry-v2 {
+  server registryv2:5000;
+}
+
+
+server {
+  listen 5441;
+  server_name localhost;
+  ssl on;
```

Comment:
nginx manual recommends to use ```ssl``` option of ```listen``` directive instead of  ```ssl on``` : ```listen 5441 ssl;```. I don't know why, but it does :)
Also it's good to use SSL cache (http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_session_cache):  ```ssl_session_cache   shared:SSL:10m;```